File: Api/Timesheet/Entities/MenuItemModel.cs
namespace Timesheet.Entities
{
    public class MenuItemModel
    {
        public int? id { get; set; }
        public int? idFav { get; set; }
        public string title { get; set; }
        public string target { get; set; }
        public string iconClass { get; set; }
        public int? parentId { get; set; }
        public bool isFavority { get; set; }
        public string permission { get; set; }
        public bool isGroup { get; set; }
    }
}

File: Web/src/lib/models/menu-item.model.ts
export class MenuItemModel {
  id: string | number;
  idFav: string | number;
  title: string;
  target: string;
  iconClass: string;
  parentId: string | number;
  isFavority: boolean;
  permission: string;

  constructor(obj?: MenuItemModel) {
    Object.assign(this, obj);
  }
}
